### ISSUE TITLE
Add entries to things from Miscoroft Store to Windows Terminal.

### ISSUE THREAD

**Issue by AraHaan** [microsoft/terminal#8732]
# Description of the new feature/enhancement
I would like for Windows Terminal to list other console programs installed from the microsoft store (that provide interface like powershell, bash, command prompt, etc), I would also like for the default windows powershell to be replaced as an option if an microsoft store version is installed as well, e.g they installed powershell stable from ms store, or powershell preview to replace the one originally built into windows. Or it could be a form of the user actually configuring what they want to be used as tab options when making a new terminal tab instead as well too where they can say ``this is from the Microsoft Store``, then they can select the program from a list of those installed from the Microsoft Store, and then so on until they list all the options they want to have (they could then configure to add things like git bash, etc that is installed from git for windows too from that configuration step then if taken that route too which would be great).

# Proposed technical implementation details (optional)
Basically I want for console applications that provides interface (like command prompt, powerhsell, bash, etc) to show up as possible options on the tabs in terminal, as well as if ms store version of powershell installed for it to use that one instead of the one built into windows itself.

**Comment by KalleOlaviNiemitalo** [comment]
See the [For apps installed through Microsoft store (or similar)](https://github.com/microsoft/terminal/blob/64666b22fba7584158cd1029d47083d5b59a212b/doc/specs/Proto%20extensions-spec.md#for-apps-installed-through-microsoft-store-or-similar) section from <https://github.com/microsoft/terminal/pull/7584>, for information about how the apps would be able to define profiles for Microsoft Terminal. The implementation <https://github.com/microsoft/terminal/pull/7632> has not been merged in yet.

Somewhat related: <https://github.com/microsoft/terminal/issues/1394> (Git Bash profile), <https://github.com/microsoft/terminal/issues/3821> (Visual Studio profiles), <https://github.com/microsoft/terminal/issues/4000> (extensions in general).

**Comment by DHowett** [comment]
@KalleOlaviNiemitalo covered this robustly as a duplicate of a bunch of our current and future plans, so I'm going to close this one out.

At the end of the day, though, we've distancing ourselves from the detection model and putting more control in the hands of applications such that those applications can tell us where, when and how to launch them.
